metadatastore/core.py

```diff
@@ -1043,7 +1044,7 @@ def find_events(start_col, start_cache,
         # wrap it our fancy dict
         ev = doc.Document('Event', ev)
         yield ev
-
+    events.close()
```

Comment:
Can you wrap this whole thing in a `try...finally` or is there a context manager for this sort of thing (quick google search says no).